src/dmd/expressionsem.d

```diff
@@ -6922,6 +6922,18 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             exp.error("cannot take address of `%s`", exp.e1.toChars());
             return setError();
         }
+        if (auto dve = exp.e1.isDotVarExp())
+        {
+            /* https://issues.dlang.org/show_bug.cgi?id=22749
+             * Error about taking address of any bit-field, regardless of
+             * whether SCOPE.Cfile is set.
+             */
+            if (auto bf = dve.var.isBitFieldDeclaration())
+            {
+                exp.error("cannot take address of bit-field `%s`", bf.toChars());
```

Comment:
Supplemental error too?

Reply:
i.e. print where the bitfield is defined since you could be consuming C code from who knows where

Reply:
Why? You're taking the address of a bit-field, enough said.

Reply:
I guess, just that context as to where the bitfield came from doesn't hurt